v2/util/algorithm_identifier.go

```diff
@@ -157,3 +157,28 @@ func GetPublicKeyOID(c *x509.Certificate) (asn1.ObjectIdentifier, error) {
 
 	return publicKeyOID, nil
 }
+
+// Returns the algorithm field of the SubjectPublicKeyInfo of the certificate in its encoded form (containing Tag
+// and Length) or an error if the algorithm field could not be extracted.
+//
+//    SubjectPublicKeyInfo  ::=  SEQUENCE  {
+//        algorithm            AlgorithmIdentifier,
+//        subjectPublicKey     BIT STRING  }
+//
+func GetPublicKeyAidEncoded(c *x509.Certificate) ([]byte, error) {
```

Comment:
If you're up for it I think `lint_spki_rsa_encryption_parameter_not_null.go` could have it's `Execute` simplified by calling this new utility function.

Reply:
Done